distributed/protocol/serialize.py

```diff
@@ -54,19 +54,30 @@ def dask_loads(header, frames):
 
 
 def pickle_dumps(x, context=None):
-    header = {"serializer": "pickle"}
     frames = [None]
     buffer_callback = lambda f: frames.append(memoryview(f))
     frames[0] = pickle.dumps(
         x,
         buffer_callback=buffer_callback,
         protocol=context.get("pickle-protocol", None) if context else None,
     )
+    header = {
+        "serializer": "pickle",
+        "pickle-writeable": tuple(not f.readonly for f in frames[1:]),
+    }
```

Comment:
I am not sure if we want to do that here or in the individual registered dumps/loads functions like the numpy serialization does?
Anyways, I don't think it should block this PR.

Reply:
Yeah it's a good question. I think support for NumPy arrays is a bit older as it is a primary use case. So that function may just be a bit unusual because of that.

We should be ok pulling this out of the NumPy case and handling them generally. I would think that should yield simpler easier to understand code, but could be wrong about that

For context tracking writeable frames was needed to solve some gnarly issues ( https://github.com/dask/distributed/issues/1978 ) ( https://github.com/dask/distributed/issues/3943 ). So if there is a general way to solve this, that would be ideal to ensure they don't resurface

Reply:
I agree but let's do that in a follow up PR. 
It assumes that `dask_dumps` returns a `memoryview` compatible object, is that right?
Also, we apparently allow additionally frames when deserializing: https://github.com/dask/distributed/blob/master/distributed/protocol/tests/test_serialize.py#L82

Reply:
Sure sounds good 🙂

Yeah though I think that is pretty closely enforced today

I think that is just showing we ignore empty frames, but could be missing something